03week/towersOfHanoi.js

```diff
@@ -18,28 +22,52 @@ function printStacks() {
   console.log("b: " + stacks.b);
   console.log("c: " + stacks.c);
 }
-
-function movePiece() {
-  // Your code here
-
+/*The stacks object has 3 keys (a, b, c) and the key values 4, 3, 2, 1 are in an array.
+The .pop method will be used to remove the last element from the stacks array.
+Then the .push method will be used to place the removed value (block) that was popped
+and place it in any of the other stacks.*/
+const movePiece = (startStack, endStack) => {
+  stacks[endStack].push(stacks[startStack].pop());
 }
 
-function isLegal() {
+/*A move is legal is when a smaller block is placed in an empty stack by itself,
+ if a stack is occupied by a larger block then the move is not allowed. The .length
+ method is used to make the occupied stack equal to 0. */
+const isLegal = (startStack, endStack) => {
   // Your code here
-
+  if (stacks[endStack].length === 0) {
+    return true;
+  } else if (stacks[startStack].pop() < stacks[endStack].pop()) {
+    return true;
+  } else {
+    return false;
+  }
```

Comment:
This whole if/else could be simplified to `return stacks[endStack].length === 0 || stacks[startStack].pop() < stacks[endStack].pop()`